adal/src/main/java/com/microsoft/aad/adal/Discovery.java

```diff
@@ -191,6 +194,15 @@ public void setCorrelationId(final UUID requestCorrelationId) {
     }
 
     /**
+     * Set the context for the tenant discovery call.
+     *
+     * @param context
+     */
+    void setContext(final Context context) {
```

Comment:
What happens if this setter wasn't called and someone invokes `Discovery#performInstanceDiscovery`?

Reply:
Discovery is an internal class in adal and the Context could not passed into the constructor, So we need to call the setter whenever we create a Discovery instance.